srcCpp/RTIRawTransportImpl.cxx

```diff
@@ -0,0 +1,1481 @@
+            }
+
+            if (_peersMap.size() == 0) {
+                fprintf(stderr, "Any peer correspond to a valid address\n");
+                return false;
+            }
+
+            break;
+    }
+
+    case TRANSPORT_SHMEM:
+    {
+        struct NDDS_Transport_Shmem_Property_t shmem_prop =
+                NDDS_TRANSPORT_SHMEM_PROPERTY_DEFAULT;
+
+        //TODO: move to validate function when parameter manager merge
+        if (!_useBlocking) {
+            fprintf(stderr, "SHMEM dont support -noBlockingSockets\n");
+            return false;
+        }
+
+        /* _Plugin configure for shared memory */
+        shmem_prop.parent.message_size_max =
+                NDDS_TRANSPORT_UDPV4_PAYLOAD_SIZE_MAX;
+        shmem_prop.receive_buffer_size =
+                NDDS_TRANSPORT_UDPV4_PAYLOAD_SIZE_MAX * 10;
+        shmem_prop.received_message_count_max =
+                shmem_prop.receive_buffer_size / perftest_cpp::OVERHEAD_BYTES;
+        /*
+         * The total number of bytes that can be buffered in the receive
+         * queue is calculated with the following formula:
+         * size = receive_buffer_size + message_size_max +
+         *          received_message_count_max * fixedOverhead
+         */
+
+        _plugin = NDDS_Transport_Shmem_new(&shmem_prop);
+
+        if (_plugin == NULL) {
+            fprintf(stderr, "Fail to create transport plugin\n");
+            return false;
+        }
+
+        /* Peers parse to NDDS_Transport_Address_t */
+        NDDS_Transport_Address_t addr = NDDS_TRANSPORT_ADDRESS_INVALID;
+        char addr_sub[NDDS_TRANSPORT_ADDRESS_STRING_BUFFER_SIZE];
+        int id_sub = 0;
+
+        for (int i = 0; i < _peerHostCount; i++) {
+            id_sub = 0;
+            sscanf(_peerHost[i], "%[^:]:%d", addr_sub, &id_sub);
+
+            _peersMap.push_back(
+                    std::pair<NDDS_Transport_Address_t, int>(addr, id_sub));
+        }
+
+        /*
+         * For SHMEM we dont want to print any interface on the
+         * printTransportConfigurationSummary()
+         */
+        _transport.allowInterfaces = std::string();
+
+        break;
+    }
+
+    case TRANSPORT_UDPv6:
+    case TRANSPORT_TCPv4:
+    case TRANSPORT_TLSv4:
+    case TRANSPORT_DTLSv4:
+    case TRANSPORT_WANv4:
+    default:
+        fprintf(stderr, "RawTransport only support UDPv4 & SHMEM\n");
+        return false;
+
+    } /* End Switch */
+
+
+
+    _transport.printTransportConfigurationSummary();
+
+    delete interface;
+
+    return true;
+}
+
+int getNumMulticastInterfaces(struct NDDS_Transport_UDP *plugin)
+{
+    int count = 0;
+
+    if (plugin == NULL) {
+        return count;
+    }
+
+    for (int i = 0; i < plugin->_interfacesCount; i++) {
+        if (plugin->_interfaceArray[i]._interfaceFlags
+            & RTI_OSAPI_SOCKET_INTERFACE_FLAG_MULTICAST) {
+            ++count;
+        }
+    }
+    return count;
+}
+
+struct REDAWorker *RawTransport_get_worker_per_threadI(
```

Comment:
Change name, RawTrasnportGetWorkerPerThread

Discuss with Jose and Javi Worker creations.

Reply:
Name fixed.